integration-tests/src/test/java/com/datastax/oss/driver/mapper/ProfileIT.java

```diff
@@ -0,0 +1,295 @@
+
+    ProfileIT.InventoryMapper inventoryMapper =
+        new ProfileIT_InventoryMapperBuilder(mapperSession).build();
+    dao = inventoryMapper.simpleDao("cl");
+  }
+
+  @Before
+  public void setup() {
+    SIMULACRON_RULE.cluster().clearLogs();
+  }
+
+  private static final ProfileIT.Simple simple = new ProfileIT.Simple(UUID.randomUUID(), "DATA");
+
+  @Test
+  public void should_honor_exec_profile_on_insert() {
+    dao.save(simple);
+
+    ClusterQueryLogReport report = SIMULACRON_RULE.cluster().getLogs();
+    validateQueryOptions(report.getQueryLogs().get(0));
+  }
+
+  @Test
+  public void should_honor_exec_profile_on_delete() {
+    dao.delete(simple);
+
+    ClusterQueryLogReport report = SIMULACRON_RULE.cluster().getLogs();
+    validateQueryOptions(report.getQueryLogs().get(0));
+  }
+
+  @Test
+  public void should_honor_exec_profile_on_update() {
+    dao.update(simple);
+
+    ClusterQueryLogReport report = SIMULACRON_RULE.cluster().getLogs();
+    validateQueryOptions(report.getQueryLogs().get(0));
+  }
+
+  @Test
+  public void should_honor_exec_profile_on_query() {
+    dao.findByPk(simple.pk);
+
+    ClusterQueryLogReport report = SIMULACRON_RULE.cluster().getLogs();
+    validateQueryOptions(report.getQueryLogs().get(0));
+  }
+
+  private void validateQueryOptions(QueryLog log) {
+
+    Message message = log.getFrame().message;
+    assertThat(message).isInstanceOf(Execute.class);
+    Execute queryExecute = (Execute) message;
+    assertThat(queryExecute.options.consistency)
+        .isEqualTo(DefaultConsistencyLevel.ANY.getProtocolCode());
+  }
+
+  private static void primeInsertQuery() {
+    Map<String, Object> params = ImmutableMap.of("pk", simple.getPk(), "data", simple.getData());
+    Map<String, String> paramTypes = ImmutableMap.of("pk", "uuid", "data", "ascii");
+    SIMULACRON_RULE
+        .cluster()
+        .prime(
+            when(query(
+                    "INSERT INTO ks.simple (pk,data) VALUES (:pk,:data)",
+                    Lists.newArrayList(
+                        com.datastax.oss.simulacron.common.codec.ConsistencyLevel.ONE,
+                        com.datastax.oss.simulacron.common.codec.ConsistencyLevel.ANY),
+                    params,
+                    paramTypes))
+                .then(noRows()));
+  }
+
+  private static void primeDeleteQuery() {
+    Map<String, Object> params = ImmutableMap.of("pk", simple.getPk());
+    Map<String, String> paramTypes = ImmutableMap.of("pk", "uuid");
+    SIMULACRON_RULE
+        .cluster()
+        .prime(
+            when(query(
+                    "DELETE FROM ks.simple WHERE pk=:pk",
+                    Lists.newArrayList(
+                        com.datastax.oss.simulacron.common.codec.ConsistencyLevel.ONE,
+                        com.datastax.oss.simulacron.common.codec.ConsistencyLevel.ANY),
+                    params,
+                    paramTypes))
```

Comment:
It might look that way at first glance, but I think if you look again you will see this is more trouble than it's worth. Parameters and return types change